Corrfunc/mocks/DDtheta_mocks.py

```diff
@@ -351,6 +353,391 @@ def DDtheta_mocks(autocorr, nthreads, binfile,
+        is a tuple containing ((nRA, nDEC), runtimes). ``runtimes`` is a
+        ``numpy`` structured array containing the fields, [``nRA``, ``nDEC``,
+        ``avg_runtime``, ``sigma_time``]. Here, ``avg_runtime`` is the
+        average time, measured over ``nrepeats`` invocations, spent in
+        the python extension. ``sigma_time`` is the dispersion of the
+        run times across those ``nrepeats`` invocations.
+
+    Example
+    -------
+    >>> from __future__ import print_function
+    >>> import numpy as np
+    >>> import time
+    >>> from math import pi
+    >>> from os.path import dirname, abspath, join as pjoin
+    >>> import Corrfunc
+    >>> from Corrfunc.mocks.DDtheta_mocks \
+        import find_fastest_DDtheta_mocks_bin_refs
+    >>> binfile = pjoin(dirname(abspath(Corrfunc.__file__)),
+    ...                 "../mocks/tests/", "angular_bins")
+    >>> N = 100000
+    >>> nthreads = 4
+    >>> seed = 42
+    >>> np.random.seed(seed)
+    >>> RA1 = np.random.uniform(0.0, 2.0*pi, N)*180.0/pi
+    >>> cos_theta = np.random.uniform(-1.0, 1.0, N)
+    >>> DEC1 = 90.0 - np.arccos(cos_theta)*180.0/pi
+    >>> autocorr = 1
+    >>> best, _ = find_fastest_DDtheta_mocks_bin_refs(autocorr, nthreads, \
+    ...                                               binfile, RA1, DEC1, \
+    ...                                               return_runtimes=True)
+    >>> print(best) # doctest:+SKIP
+    (2, 1)
+
+    .. note:: Since the result might change depending on the computer,
+    doctest is skipped for this function.
+    """
+
+    weights1 = None
+    weights2 = None
+    weight_type = None
+
+    try:
+        from Corrfunc._countpairs_mocks import countpairs_theta_mocks as \
+            DDtheta_mocks_extn
+    except ImportError:
+        msg0 = "Could not import the C extension for the angular " \
+              "correlation function for mocks."
+        raise ImportError(msg0)
+
+    import numpy as np
+    from Corrfunc.utils import translate_isa_string_to_enum, fix_ra_dec, \
+        return_file_with_rbins, convert_to_native_endian, process_weights
+    from future.utils import bytes_to_native_str
+    import itertools
+    import time
+
+    if autocorr == 0:
+        if RA2 is None or DEC2 is None:
+            msg1 = "Must pass valid arrays for RA2/DEC2 for " \
+                  "computing cross-correlation."
+            raise ValueError(msg1)
+    else:
+        RA2 = np.empty(1)
+        DEC2 = np.empty(1)
+
+    if link_in_ra is True:
+        link_in_dec = True
+
+    if link_in_dec is False and link_in_ra is False:
+        msg2 = "Warning: Brute-force calculation without any gridding " \
+               "is forced, as link_in_dec and link_in_ra are both set " \
+               "to False. Please be sure to turn on at least link_in_dec, " \
+               "or both link_in_dec and link_in_ra on."
+        raise ValueError(msg2)
+
+    if link_in_dec is True:
```

Comment:
You have previously set (line 537-538) ``link_in_dec=True`` when ``link_in_ra=True`` -- which means you can replace the two conditions with ``if link_in_ra:``